### PR TITLE
Fix plugin auto search dirs

### LINKED ISSUE #8474: Plugin autoload doesn't work for packages installed from GitHub and for Yarn PnP
**Environments:**

- Prettier Version: 2.0.5
- Running Prettier via: CLI
- Runtime: Node.js v14
- Operating System: Windows

**Steps to reproduce:**

```console
yarn add prettier/prettier @prettier/plugin-pug
```

**Expected behavior:**

Can format pug file with `npx prettier test.pug`

**Actual behavior:**

The CLI can't load `pug` plugin automatically

----

Problem should be [here](https://github.com/prettier/prettier/blob/6b4c4066aa1d59e6f806b1646beee314f93103d3/src/common/load-plugins.js#L30), when install from github, `./node_modules/prettier` maybe has own `node_modules` dir, so the `autoLoadDir` will be `./node_modules/prettier`, not expected `./`, so we are looking for `./node_modules/prettier/node_modules/@prettier/plugin-*` not `./node_modules/@prettier/plugin-*` (where the `@prettier/plugin-pug` is installed).

### PR DESCRIPTION
**Pull request by fisker**
Fixes #8474

Not sure how to test it

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/load-plugins.js [hunk 1]
```diff
@@ -5,18 +5,43 @@ const partition = require("lodash/partition");
...
+    pluginSearchDirs.push(autoLoadDir);
+  }
+
+  // We may have plugins in `prettier/prettier` codebase for test
+  if (isDirectory(path.join(packageRoot, "node_modules"))) {
+    pluginSearchDirs.push(packageRoot);
+  }
+
+  return pluginSearchDirs;
+}
```
Comment (evilebottnawi) [inline src/common/load-plugins.js:43]: What about yarn pnp?

Comment (fisker) [reply src/common/load-plugins.js:43]: We don't support it.


Possible solution `--plugin=@prettier/plugin-pug`

Comment (evilebottnawi) [reply src/common/load-plugins.js:43]: I think will be great to support them in the future

Comment (thorn0) [reply src/common/load-plugins.js:43]: @evilebottnawi I'm interested in your thoughts on https://github.com/prettier/prettier/issues/8474#issuecomment-639013515.

**Comment by fisker** [comment]
Can we merge this first, to make [this](https://github.com/prettier/plugin-pug/issues/5#issuecomment-658651088) work, without additional cli argument?

File: src/common/load-plugins.js (hunk 1 above)
Comment (merceyz) [reply src/common/load-plugins.js:43]: You shouldn't have to reinvent the node resolution, why not use `require.resolve` to find the plugin from the project that depends on prettier / whatever invokes it?

Comment (fisker) [reply src/common/load-plugins.js:43]: Because we don't know the plugin name yet.

Comment (merceyz) [reply src/common/load-plugins.js:43]: Oh, I assumed you loaded the name from `package.json` which would be the sensible thing to do

Comment (fisker) [reply src/common/load-plugins.js:43]: Currently, we search `node_modules`, so some plugins might be "indirect" dependencies.

Comment (fisker) [reply src/common/load-plugins.js:43]: Imaging this, you have `@merceyz/prettier` package, include `config` and `plugins`, you may don't want list all plugins in every project.

Comment (merceyz) [reply src/common/load-plugins.js:43]: Then I would assume `@merceyz/prettier` declares the plugins name/location in a config and it's loaded from there, basically the same way babel handles plugins/presets

Comment (fisker) [reply src/common/load-plugins.js:43]: Yep, that's my proposal https://github.com/prettier/prettier/issues/8474#issuecomment-661543282, but we haven't support it.

**Comment by fisker** [comment]
Plugin search already been removed #14759

**Pull request closed without merging**